src/utils.ts

```diff
@@ -823,3 +831,247 @@ export const messageSetPagination = <StreamChatGenerics extends ExtendableGeneri
     return messagePaginationLinear(params);
   }
 };
+
+/**
+ * A utility object used to prevent duplicate invocation of channel.watch() to be triggered when
+ * 'notification.message_new' and 'notification.added_to_channel' events arrive at the same time.
+ */
+const WATCH_QUERY_IN_PROGRESS_FOR_CHANNEL: Record<string, Promise<QueryChannelAPIResponse> | undefined> = {};
+
+type GetChannelParams<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> = {
+  client: StreamChat<StreamChatGenerics>;
+  channel?: Channel<StreamChatGenerics>;
+  id?: string;
+  members?: string[];
+  options?: ChannelQueryOptions<StreamChatGenerics>;
+  type?: string;
+};
+/**
+ * Calls channel.watch() if it was not already recently called. Waits for watch promise to resolve even if it was invoked previously.
+ * If the channel is not passed as a property, it will get it either by its channel.cid or by its members list and do the same.
+ * @param client
+ * @param members
+ * @param options
+ * @param type
+ * @param id
+ * @param channel
+ */
+export const getAndWatchChannel = async <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
```

Comment:
Does this come from React SDK? (https://github.com/GetStream/stream-chat-react/blob/master/src/utils/getChannel.ts)